### PR TITLE
Merge feature/mlbeta2 into master

### PR DESCRIPTION
**Pull request by yebrahim**
Branch has been rebased to latest master.

### CODE REVIEW

File: containers/base/Dockerfile [hunk 1]
```diff
@@ -31,7 +31,7 @@ RUN echo "deb-src http://ftp.us.debian.org/debian testing main" >> /etc/apt/sour
 # Save GPL source packages
 RUN mkdir -p /srcs && \
     cd /srcs && \
-    apt-get source -d python-zmq wget git ca-certificates pkg-config libpng-dev && \
+    apt-get source -d python-zmq wget ca-certificates pkg-config libpng-dev && \
```
Comment (ojarjur) [inline containers/base/Dockerfile]: We should remove this line of change; we still need the git source code.

File: containers/base/Dockerfile [hunk 2]
```diff
@@ -141,7 +145,7 @@ RUN ipython profile create default && \
     if [ -d /datalab/lib/pydatalab/.git ]; then \
         echo "use local lib"; \
       else \
-        git clone https://github.com/googledatalab/pydatalab.git /datalab/lib/pydatalab; \
+        git clone https://github.com/googledatalab/pydatalab.git  --single-branch --branch feature/ml /datalab/lib/pydatalab; \
```
Comment (ojarjur) [inline containers/base/Dockerfile]: We should get rid of this line of change too; we want to keep the master branch.

File: containers/datalab/content/run.sh [hunk 3]
```diff
@@ -179,6 +179,7 @@ else
 fi
 (cd /content/datalab/docs; git config core.sparsecheckout true; echo $'intro/\nsamples/\ntutorials/\n*.ipynb\n' > .git/info/sparse-checkout; git checkout master)
 
+
```
Comment (ojarjur) [inline containers/datalab/content/run.sh]: It looks like this file got changed on accident.

Comment (yebrahim) [inline containers/datalab/content/run.sh]: fixed

**Pull request merged**